### PR TITLE
Explicit Close() when security-file-token provider writes tokens

### LINKED ISSUE #2248: Refactor security-file-token-provider's provider.go to perform explicit closes
Pertains to this comment
https://github.com/edgexfoundry/edgex-go/pull/2202#pullrequestreview-333460337

The ideal logic we want here is
```
writeCloser, err := p.fileOpener.OpenFileWriter(outputTokenFilename, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, os.FileMode(0600))
// err check here
// chmod() here
if err
  close()
  return err
// chown here
if err
  close()
  return err
// create encoder/writer
if encoder nil
  close()
  return err
// encode
if err
  close
  return err
close()
```

This is to adopt best-known-practice to avoid deferred closes when writing files.

### PR DESCRIPTION
**Pull request by bnevis-i**
This is a followup to a previous code review to clean up
the logic to explicitly Close() a file being written that
contains the authentication token to the secret store.
Deferred Close()s are removed and replaced with
explicit closes before leaving the function and an error-
reporting Close() at the end of each loop iteration.

Fixes #2248

Signed-off-by: Bryon Nevis <bryon.nevis@intel.com>

### CODE REVIEW

**Comment by bnevis-i** [comment]
This PR addresses the issues raised in https://github.com/edgexfoundry/edgex-go/pull/2202

File: internal/security/fileprovider/provider.go [hunk 1]
```diff
@@ -153,22 +153,20 @@ func (p *fileTokenProvider) Run() error {
...
-		}()
+		// writeCloser is writable file -- explicitly close() to ensure we catch errors writing to it
 
 		permissionable, ok := writeCloser.(permissionable)
 		if ok {
 			if serviceConfig.FilePermissions != nil &&
 				(serviceConfig.FilePermissions).ModeOctal != nil {
 				mode, err := strconv.ParseInt(*(serviceConfig.FilePermissions).ModeOctal, 8, 32)
 				if err != nil {
+					writeCloser.Close()
```
Comment (michaelestrin) [inline internal/security/fileprovider/provider.go]: Recommend explicitly swallowing the error return value; i.e. `_ = writeCloser.Close()`

Mentioned once, occurs elsewhere.

Comment (bnevis-i) [reply internal/security/fileprovider/provider.go]: Implemented as suggested 5X.

**Comment by bnevis-i** [comment]
Rebased with fixes.

**Pull request merged**